pom.xml

```diff
@@ -176,7 +176,7 @@
           <execution>
             <phase>verify</phase>
             <goals>
-              <goal>check</goal>
+              <goal>format</goal>
```

Comment:
Please do NOT change this. If you want to format code, you can run `mvn fmt:format` as described in CONTRIBUTING.md. :)

Reply:
FYI, the reason to not change this is because Travis CI depends on the `mvn fmt` command working this way to check that people have properly formatted their code before check-in.

Reply:
If you leave it at check, then the build fails. I hope nobody is checking in code when the build failed. If your goal is to have the code formatted, why not have formatting be a part of every build? What am I missing here?

Reply:
In theory, no one should be checking in code when the build failed, because we have the Travis check to forestall that (at least for the common.graph and master branches).

I haven't formed a strong opinion on this issue yet, but I think that the point of leaving it as a check is to not require formatting on every build; I can imagine that there might be circumstances in which you might want to leave the formatting until you were done developing.

But maybe there's something I'm missing; @jbduncan, can you elaborate?

Reply:
@jrtom I admit I'm not too clear either on why checking is preferred to formatting at CI-time. I think it's a combination of reasons.

I can think of two reasons ATM. Firstly, it's partially to do with how formatters (google-java-format included) can sometimes contain bugs which can cause code to be formatted weirdly, or to flat-out fail with an error or exception. So, if formatting happened during CI-time, the CI may fail or format things strangely for reasons not to do with the checked-in code, which has the potential to make things confusing.

Secondly, formatters are prone to occasionally formatting code incompletely, thus requiring formatters to be run twice or so, so formatting by CI wouldn't be hermetic, which again is a cause for confusion. I actually experienced this "problem" (if you'd even call it that) with google-java-format when I was working on introducing it to JUNG a good few weeks/months ago; it required me to run `mvn fmt:format` twice before certain source files were formatted completely, strangely.

I hope @nedtwigg, who I collaborate with on [Spotless](https://github.com/diffplug/spotless), can elaborate further on this, because I'm sure an issue was opened in the past on Spotless's issue tracker asking why Spotless's `spotlessCheck` task (similar to `mvn fmt:format` for us) is the default over `spotlessApply` (similar to `mvn fmt:check`), but I can't find that issue yet.

Reply:
Replying to https://github.com/jrtom/jung/pull/126#issuecomment-333177281:

@tomnelson That sounds good to me, having a default maven profile for formatting code and a CI-specific one for checking code.

Would you be happy to create a new PR for that?

If not, I may have the time and energy in the foreseeable future to investigate Maven profiles and create them myself.

Reply:
We definitely wouldn't want the CI check to be formatting code itself; I don't want an automated CI check to be trying to modify the code (I don't even know what that would do, but I feel no strong need to find out :) ).

So I would say that until we have multiple profiles that we should keep it at "check".

Reply:
Replying to comment https://github.com/jrtom/jung/pull/126#issuecomment-333180321: 

@tomnelson It should be easy to apply the new profile without needing admin permissions. I assume that profiles can be chosen from the command-line? If so, all you'd need to do is change [this line](https://github.com/jrtom/jung/blob/master/.travis.yml#L19) in the Travis config to ensure it runs the right profile.